src/GitHub.Api/SimpleApiClient.cs

```diff
@@ -66,7 +66,12 @@ public class SimpleApiClient : ISimpleApiClient
                     }
                 }
             }
-            // it'll throw if it's private
+            // it'll throw if it's private or an enterprise instance requiring authentication
+            catch (ForbiddenException)
+            {
+                if (!HostAddress.IsGitHubDotComUri(OriginalUrl.ToRepositoryUrl()))
```

Comment:
Yeah, flow analysis is completely wrong here, this exception can only be thrown inside the condition that guarantees that `OriginalUrl` is not null (and `OriginalUrl` is readonly it's impossible for it to be changed in a way that makes it non-null to go into the condition and null in the catch block). I'm doing some changes to the constructor so this it not an issue anymore.